snapcraft/plugins/copy.py

```diff
@@ -65,6 +67,13 @@ def schema(cls):
 
         return schema
 
+    def __init__(self, name, options, project):
+        super().__init__(name, options, project)
+
+        logger.warning("DEPRECATED: The 'copy' plugin's functionality "
+                       "has been replaced by the 'dump' plugin, and it will "
```

Comment:
I think that's too much info for a deprecation notice. If they want to learn more they can look at the docs for that plugin.